commands/audit/audit.go

```diff
@@ -194,10 +195,18 @@ func RunAudit(auditParams *AuditParams) (results *utils.Results, err error) {
 		return
 	}
 	results.XrayVersion = auditParams.xrayVersion
+
 	results.ExtendedScanResults.EntitledForJas, err = isEntitledForJas(xrayManager, auditParams)
 	if err != nil {
 		return
 	}
+	dynamicTokenVersionMismatchErr := clientutils.ValidateMinimumVersion(clientutils.Xray, auditParams.xrayVersion, jasutils.DynamicTokenValidationMinXrayVersion)
+	if dynamicTokenVersionMismatchErr != nil && (auditParams.AuditBasicParams.ValidateSecrets) {
+		log.Warn("Token validation (--validate-secrets flag) is not supported in your xray version")
+		results.ExtendedScanResults.SecretValidation = false
+	} else {
+		results.ExtendedScanResults.SecretValidation = jas.CheckForSecretValidation(xrayManager, auditParams.AuditBasicParams.ValidateSecrets)
+	}
```

Comment:
to reduce code duplication and to make sure this is always running, put clientutils.ValidateMinimumVersion inside of CheckForSecretValidation right before running the API call.
(also true for the one in scan)